src/core/security/client_auth_filter.c

```diff
@@ -199,8 +200,28 @@ static void auth_start_transport_op(grpc_call_element *elem,
   channel_data *chand = elem->channel_data;
   grpc_linked_mdelem *l;
   size_t i;
+  grpc_client_security_context* sec_ctx = NULL;
 
-  /* TODO(jboeuf): write the call auth context. */
+  if (calld->security_context_set == 0) {
+    calld->security_context_set = 1;
+    GPR_ASSERT(op->context);
+    if (op->context[GRPC_CONTEXT_SECURITY].value == NULL) {
+      op->context[GRPC_CONTEXT_SECURITY].value =
+          grpc_client_security_context_create();
+      op->context[GRPC_CONTEXT_SECURITY].destroy =
+          grpc_client_security_context_destroy;
+    }
+    sec_ctx = op->context[GRPC_CONTEXT_SECURITY].value;
+    if (sec_ctx->auth_context == NULL) {
+      sec_ctx->auth_context =
+          GRPC_AUTH_CONTEXT_REF(chand->security_connector->base.auth_context,
+                                "client_auth_filter");
+    } else {
+      sec_ctx->auth_context->chained =
```

Comment:
OK I understand now. Sorry about that... In this case, on the client side, the auth_context that will always be the one from the connector (i.e. transport security). We have chained contexts in general for a context issued from metadata processing containing a (chained) context from transport security which happens on the server side. 

So I would simplify the code here to (starting line 214):

```
sec_ctx = op->context[GRPC_CONTEXT_SECURITY].value;
GRPC_AUTH_CONTEXT_UNREF(sec_ctx->auth_context, "client auth filter");
sec_ctx->auth_context = GRPC_AUTH_CONTEXT_REF(chand->security_connector->base.auth_context, "client_auth_filter");
```
Does that make sense? BTW, thanks much for tackling this part of the implementation that I should have done.